### PR TITLE
[tsp-client] Simplify local spec scenario for `init` command

### LINKED ISSUE #9261: [tsp-client] tsp-client init --local-spec-repo flag not working
Running the command:

`tsp-client init -c <URL> --local-spec-repo <PATH TO LOCAL>`

Any changes I make to my local tspconfig.yaml are not being used. I must commit and push the changes then update my URL.

Oddly enough, if I have an error in my local tspconfig.yaml, an error will be thrown when generating.

So somehow the command is throwing an error when the local project isn't valid, but it's also not using any changes made in my local project.

### PR DESCRIPTION
**Pull request by catalinaperalta**
Fixes #9261
Part of https://github.com/Azure/azure-sdk-tools/issues/8639

### CODE REVIEW

File: tools/tsp-client/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Release
 
+## Unreleased - 0.14.0
+
+- Fix `init` command when using a local spec:
```
Comment (timotheeguerin) [inline tools/tsp-client/CHANGELOG.md:5]: alignment issue?

File: tools/tsp-client/test/commands.spec.ts [hunk 2]
```diff
@@ -202,4 +203,23 @@ describe("Verify commands", async function () {
       assert.fail("Failed to init. Error: " + error);
     }
   });
+
+  await it("Init with local spec", async function () {
```
[the remark this answers is not in the document]
Comment (catalinaperalta) [reply tools/tsp-client/test/commands.spec.ts:207]: Eventually have to figure out a better solution but we needed tsp-client tests to check for all changes in the tool

File: tools/tsp-client/CHANGELOG.md (hunk 1 above)
Comment (catalinaperalta) [reply tools/tsp-client/CHANGELOG.md:5]: Did it on purpose, let me check if it looks good

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**